### PR TITLE
XWIKI-21988: Page naming strategy is blocking creation of pages not respecting it

### PR DESCRIPTION
**Pull request by surli**
# Jira URL

https://jira.xwiki.org/browse/XWIKI-21988

# Changes

## Description

The goal here is to immediately display a feedback in the create page dialog telling the user that the reference is wrong before they click on create.

I performed following changes here:
  * Refactor CreateActionRequestHandler to have a clean separation between retrieving the reference and performing checks over templates
  * Refactor a bit CreateAction to perform immediate checks over the rights and the name strategy before even checking if the template is chosen: the goal being here to immediately provide exception if there's a problem
  * Refactor createinline.vm to inject bootstrap modal html elements when used with an ajax request: the idea here is to display an ajax modal instead of the old deprecated xwiki dialog
  * Refactor xwiki.js to use a jquery ajax call to request and display createinline
  * Add listener on xwiki:dom:updated events in both create.js and select.js to properly load them in case of dom updates

## Clarifications

*

# Screenshots & Video

I took screenshots in the exact same scenario with a link not respecting the name strategy.

### With the create modal

Capture d’écran du 2024-06-25 18-52-27

### With the create page

Capture d’écran du 2024-06-25 18-52-33

### With the create modal

Capture d’écran du 2024-06-25 18-47-31

### With the create page

Capture d’écran du 2024-06-25 18-48-27

# Executed Tests

TBD

# Expected merging strategy

* Prefers squash: Yes 
* Backport on branches:
  * 15.10.x
  * 16.4.x

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-api/src/main/resources/ApplicationResources.properties [hunk 1]
```diff
@@ -17,5 +17,9 @@
 # Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 # 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 # ---------------------------------------------------------------------------
+entitynamevalidation.create.invalidnameError=Cannot create document {0} because its name does not respect the configured name strategy of the wiki.
+entitynamevalidation.create.invalidname.possiblesolution=Change the name or move the parent to another space.
+entitynamevalidation.create.invalidname.possiblesolutionForLink=Edit the link of current page so that it complies to the current name strategy.
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-api/src/main/resources/ApplicationResources.properties:22]: I think it's `complies with`.

File: xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-api/src/main/resources/ApplicationResources.properties [hunk 2]
```diff
@@ -17,5 +17,9 @@
...
+entitynamevalidation.create.invalidname.strategyInfo=The configured name strategy is: {0}. This strategy works as defined in the following: {1}
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-api/src/main/resources/ApplicationResources.properties:23]: "in the following" what? it doesn't seem clear.

ideas:
* `as defined by the following rule: `
* `The strategy is defined as follows:`

File: xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-api/src/main/resources/ApplicationResources.properties [hunk 3]
```diff
@@ -17,5 +17,9 @@
...
 entitynamevalidation.create.invalidname=Cannot create document {0} because its name does not respect the name strategy of the wiki. Change the name or move the parent to another space.
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-api/src/main/resources/ApplicationResources.properties:24]: s/document/page?

File: xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-default/src/main/resources/ApplicationResources.properties [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this software; if not, write to the Free
+# Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+# 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+# ---------------------------------------------------------------------------
+entitynamevalidation.ReplaceCharacterEntityNameValidation.name=Character replacement
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-default/src/main/resources/ApplicationResources.properties:20]: are you sure we use uppercases in translation keys as the first letter of a part?

File: xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-api/src/main/resources/ApplicationResources.properties [hunk 5]
```diff
@@ -17,5 +17,9 @@
 # Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 # 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 # ---------------------------------------------------------------------------
+entitynamevalidation.create.invalidnameError=Cannot create document {0} because its name does not respect the configured name strategy of the wiki.
```
[the remark this answers is not in the document]
Comment (vmassol) [reply xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-api/src/main/resources/ApplicationResources.properties:20]: BTW aren't we supposed to use camelcase for translation parts?

File: xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-api/src/main/resources/ApplicationResources.properties [hunk 6]
```diff
@@ -17,5 +17,9 @@
...
+entitynamevalidation.create.invalidname.possiblesolution=Change the name or move the parent to another space.
+entitynamevalidation.create.invalidname.possiblesolutionForLink=Edit the link of current page so that it complies to the current name strategy.
+entitynamevalidation.create.invalidname.strategyInfo=The configured name strategy is: {0}. This strategy works as defined in the following: {1}
 entitynamevalidation.create.invalidname=Cannot create document {0} because its name does not respect the name strategy of the wiki. Change the name or move the parent to another space.
 entitynamevalidation.nametransformation.error=Error while transforming the page title to a correct page name: the exact page title will be used.
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-api/src/main/resources/ApplicationResources.properties:25]: So for me this should be: `entityNameValidation.nameTransformation.error`. xwiki.org is down but in the dev practices we have some rules defined.

File: xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-default/src/main/resources/ApplicationResources.properties (hunk 4 above)
Comment (surli) [reply xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-default/src/main/resources/ApplicationResources.properties:20]: Note that this translation is actually a move of an existing translation, so I can't really change its key. Also the part in camelcase is actually a component hint, I just didn't applied any transformation on it.

File: xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-api/src/main/resources/ApplicationResources.properties (hunk 6 above)
Comment (surli) [reply xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-api/src/main/resources/ApplicationResources.properties:25]: Note that this one is an old one so I can't change it.

**Pull request merged**